cargo-dist/src/lib.rs

```diff
@@ -57,8 +57,88 @@ pub mod tasks;
 #[cfg(test)]
 mod tests;
 
+/// dist env test -- make sure we have everything we need for a build.
+pub fn do_env_test(cfg: &Config) -> DistResult<()> {
+    let (dist, _manifest) = tasks::gather_work(cfg)?;
+
+    let global_builds = cfg.artifact_mode == ArtifactMode::Global;
+
+    let builds = dist.config.builds;
+
+    let need_cargo_auditable = builds.cargo.cargo_auditable && !global_builds;
+    let need_cargo_cyclonedx = builds.cargo.cargo_cyclonedx && !global_builds;
+    let need_omnibor = builds.omnibor;
+    let mut need_xwin = false;
+    let mut need_zigbuild = false;
+
+    let tools = dist.tools;
+
+    for step in dist.local_build_steps.iter() {
+        // Can't require cross-compilation tools if we aren't compiling.
+        if cfg.artifact_mode == ArtifactMode::Lies {
+            break;
+        }
+
+        match step {
+            BuildStep::Cargo(step) => {
+                let cargo = tools.cargo()?;
```

Comment:
I switched to `tools.host_target`. As a note: the code still uses question mark operators in the form of `tools.host_target.parse()?` and `step.target_triple.parse()?`, but I believe if it can't parse the host _or_ build target, it makes sense to just throw an error at that point? Those don't seem recoverable.

Reply:
_As best I can tell_, the second of those should be guaranteed to succeed here: if `step.target_triple.parse()?` _would_ fail, it fails in `tasks::gather_work(cfg)?` at the start of the function anyway.

Also it requires specifying an invalid target-triple in `dist-workspace.toml` and running `dist build --target aforementioned-invalid-target`, so I think a hard error makes sense.

Reply:
Yup, I think that's fine - both of those are genuine error cases. "Cargo is missing" comes up in supported usecases, so that was more my concern.